source/postprocess/visualization/maximum_horizontal_compressive_stress.cc

```diff
@@ -47,134 +47,161 @@ namespace aspect
-            std::array<Tensor<1,dim>,dim-1 > orthogonal_directions
-              = Utilities::orthogonal_vectors(vertical_direction);
+            const Tensor<1,dim> vertical_direction = gravity / gravity.norm();
+            std::array<Tensor<1,dim>,dim-1> orthogonal_directions = Utilities::orthogonal_vectors(vertical_direction);
             for (unsigned int i=0; i<orthogonal_directions.size(); ++i)
               orthogonal_directions[i] /= orthogonal_directions[i].norm();
 
             Tensor<1,dim> maximum_horizontal_compressive_stress;
             switch (dim)
               {
-                // in 2d, there is only one horizontal direction, and
-                // we have already computed it above. give it the
-                // length of the compressive_stress (now taking into account the
-                // pressure) in this direction
                 case 2:
-                {
                   maximum_horizontal_compressive_stress = orthogonal_directions[0] *
                                                           (orthogonal_directions[0] *
-                                                           ((compressive_stress
-                                                             -
-                                                             in.pressure[q] * unit_symmetric_tensor<dim>()) *
+                                                           (avg_compressive_stress *
                                                             orthogonal_directions[0]));
                   break;
-                }
 
-                // in 3d, use the formulas discussed in the
-                // documentation of the plugin below
                 case 3:
                 {
                   const double a = orthogonal_directions[0] *
-                                   (compressive_stress *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[0]);
                   const double b = orthogonal_directions[1] *
-                                   (compressive_stress *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[1]);
-                  const double c = 2.0*orthogonal_directions[0] *
-                                   (compressive_stress *
+                  const double c = 2.0 * orthogonal_directions[0] *
+                                   (avg_compressive_stress *
                                     orthogonal_directions[1]);
 
-                  // compute the two stationary points of f(alpha)
-                  const double alpha_1 = 1./2 * std::atan2 (c, a-b);
+                  const double alpha_1 = 0.5 * std::atan2(c, a - b);
                   const double alpha_2 = alpha_1 + numbers::PI/2;
 
-                  // then check the sign of f''(alpha) to determine
-                  // which of the two stationary points is the maximum
```

Comment:
And here and all comments below. You may not think they are necessary for you, but we frequently have new students read through ASPECT's code, and generally they comment on how helpful comments like these are to understand what is happening.